go/tools/builders/compile.go

```diff
@@ -244,6 +235,11 @@ func buildImportcfgFile(archives []archive, stdImports []string, installSuffix,
 		return "", errors.New("GOROOT not set")
 	}
 	goroot = abs(goroot)
+	// UGLY HACK: The vet tool called by compile program expects the vet.cfg file
```

Comment:
By the way, do you have any thoughts about this hack? I'm not exactly sure how 'go build' gets around this (surely it has to handle cases where you run tests that do not import "fmt").

Reply:
:man_shrugging: 

It looks like go build uses the [same hack](https://github.com/golang/go/blob/master/src/cmd/go/internal/work/exec.go#L987).

My only thought about this is the `-stdimport` flags are regrettable, but I like that better than checking whether `packagefile` strings start with `GOROOT`, which is the other thing I thought of. Explicit is better than implicit.